File: output/llama3.1-8b/avatar/Python/Java/atcoder_AGC027_A.java
import java.util.Arrays;
import java.util.Scanner;

public class atcoder_AGC027_A {
    public static void main(String[] args) {
        int[] n = readIntArray();
        int x = n[0];
        int[] a = sortedArray(readIntArray());
        int ans = 0;
        if (a[0] > x) {
            System.out.println(0);
            return;
        }
        for (int i : a) {
            if (i > x) {
                x = 0;
                break;
            }
            x -= i;
            ans++;
        }
        if (x != 0) {
            ans--;
        }
        System.out.println(ans);
    }

    private static int[] readIntArray() {
        Scanner scanner = new Scanner(System.in);
        String[] input = scanner.nextLine().split(" ");
        int[] array = new int[input.length];
        for (int i = 0; i < input.length; i++) {
            array[i] = Integer.parseInt(input[i]);
        }
        return array;
    }

    private static int[] sortedArray(int[] array) {
        Arrays.sort(array);
        return array;
    }
}

File: output/llama3.1-8b_IO_1/avatar/Java/Python/atcoder_AGC027_A.py
import sys
N = int(input())
x = int(input().split()[1])
sweet = []
for _ in range(N):
    a = list(map(int, input().split()))
    sweet.append(a[0])
sweet.sort()
num = 0
while num < N and x - sweet[num] >= 0:
    x -= sweet[num]
    num += 1
if num == N and x > 0:
    num -= 1
print(num)
